alert-database/src/test/java/com/synopsys/integration/alert/database/api/DefaultProviderDataAccessorTest.java

```diff
@@ -0,0 +1,251 @@
+        ProviderUserEntity providerUserEntity = new ProviderUserEntity(projectOwnerEmail, Boolean.TRUE, 1L);
+
+        ProviderProjectRepository providerProjectRepository = Mockito.mock(ProviderProjectRepository.class);
+        ProviderUserProjectRelationRepository providerUserProjectRelationRepository = Mockito.mock(ProviderUserProjectRelationRepository.class);
+        ProviderUserRepository providerUserRepository = Mockito.mock(ProviderUserRepository.class);
+
+        Mockito.when(providerProjectRepository.findFirstByHref(Mockito.any())).thenReturn(Optional.of(providerProjectEntity));
+        Mockito.when(providerUserProjectRelationRepository.findByProviderProjectId(Mockito.any())).thenReturn(List.of(providerUserProjectRelation));
+        Mockito.when(providerUserRepository.findAllById(Mockito.any())).thenReturn(List.of(providerUserEntity));
+
+        DefaultProviderDataAccessor providerDataAccessor = new DefaultProviderDataAccessor(providerProjectRepository, providerUserProjectRelationRepository, providerUserRepository, null);
+        Set<String> emailAddresses = providerDataAccessor.getEmailAddressesForProjectHref(href);
+
+        assertEquals(1, emailAddresses.size());
+        assertTrue(emailAddresses.contains(projectOwnerEmail));
+    }
+
+    @Test
+    public void getEmailAddressesForProjectHrefEmptyTest() {
+        ProviderProjectRepository providerProjectRepository = Mockito.mock(ProviderProjectRepository.class);
+
+        Mockito.when(providerProjectRepository.findFirstByHref(Mockito.any())).thenReturn(Optional.empty());
+
+        DefaultProviderDataAccessor providerDataAccessor = new DefaultProviderDataAccessor(providerProjectRepository, null, null, null);
+        Set<String> emailAddresses = providerDataAccessor.getEmailAddressesForProjectHref("test-href");
+
+        assertEquals(0, emailAddresses.size());
+    }
+
+    @Test
+    public void getUsersByProviderConfigIdTest() {
+        Boolean optOut = Boolean.TRUE;
+
+        ProviderUserEntity providerUserEntity = new ProviderUserEntity(projectOwnerEmail, optOut, 1L);
+
+        ProviderProjectRepository providerProjectRepository = Mockito.mock(ProviderProjectRepository.class);
+        ProviderUserRepository providerUserRepository = Mockito.mock(ProviderUserRepository.class);
+
+        Mockito.when(providerUserRepository.findByProviderConfigId(Mockito.any())).thenReturn(List.of(providerUserEntity));
+
+        DefaultProviderDataAccessor providerDataAccessor = new DefaultProviderDataAccessor(providerProjectRepository, null, providerUserRepository, null);
+        List<ProviderUserModel> providerUserModelList = providerDataAccessor.getUsersByProviderConfigId(1L);
+
+        assertEquals(1, providerUserModelList.size());
+        ProviderUserModel providerUserModel = providerUserModelList.get(0);
+        assertEquals(projectOwnerEmail, providerUserModel.getEmailAddress());
+        assertEquals(optOut, providerUserModel.getOptOut());
```

Comment:
This chunk of asserts also looks like it's used in multiple places. A private method may help